### PR TITLE
Improvement: Show thumbnail, art and playcount when browsing in filemode

### PR DESCRIPTION
**Pull request by wutschel**
## Description

Before this change the app was using song icons for each file inside the Music folder, video icons for each file inside the Video/Movie/TVShow folder and picture icons for each file inside Picture folder. This was wrong. In fact, only a file icon should be shown. In case the item has a thumbnail this should be shown.

With this PR the app now will request "thumbnail", "art" (only from Kodi 11 on) and "playcount". from Kodi. The app will display these for files while browsing for music, videos, movies and TVshows. If no thumbnail or art is provided the app will fall back to folder and file icons.

Screenshots:
https://ibb.co/6n8sbKv -> Video > Files > played/unplayed video which are not in DB
https://ibb.co/m6W4c2r -> Video > Files > played video which is in DB
https://ibb.co/wS805VZ -> Music > Files > played/unplayed video which are in DB or not
https://ibb.co/X518hzr -> Music > Files > folder which hold an album which was played and is in DB
https://ibb.co/vs8yq07 -> Music > Files > song which were all played and are in DB
https://ibb.co/6g94C8r -> requesting "art" from Kodi 19.5

## Summary for release notes

Improvement: Show thumbnail, art and playcount when browsing in filemode

### CODE REVIEW

File: XBMC Remote/DetailViewController.m [hunk 1]
```diff
@@ -3364,8 +3363,13 @@ - (void)updateCellAndSaveRichData:(NSIndexPath*)indexPath watched:(int)watched i
...
     if (mutableParameters[@"file_properties"] != nil) {
-        mutableParameters[@"properties"] = mutableParameters[@"file_properties"];
-        [mutableParameters removeObjectForKey: @"file_properties"];
+        mutableParameters[@"properties"] = [mutableParameters[@"file_properties"] mutableCopy];
+        [mutableParameters removeObjectForKey:@"file_properties"];
+        
+        // Kodi 11 does not support art for file properties
+        if (AppDelegate.instance.serverVersion < 11) {
+            [mutableParameters[@"properties"] removeObject:@"art"];
+        }
```
Comment (kambala-decapitator) [inline XBMC Remote/DetailViewController.m:3372]: any chance to prevent the copy-paste? Also you could store the mutable dic in a variable to adjust it

**Review by kambala-decapitator — APPROVED** [review]
why didn't you add @3x icon?

**Comment by wutschel** [comment]
> why didn't you add @3x icon?

Just because I do not have them for the vast majority of the other icons as well. This is a todo for the future...

File: XBMC Remote/DetailViewController.m (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/DetailViewController.m]: Introduced a helper method `addFileProperties` to avoid copy-pasting.

**Comment by wutschel** [comment]
Squashed and rebased to master.

**Pull request merged**